reader/options.go

```diff
@@ -18,6 +18,19 @@ func WithNumReader(m Macro) Option {
 	}
 }
 
+// WithSymbolFactory sets the symbol factory function for the Reader.
+// This function will be used to transform a native Go string into a symbol
+// type that satisfies parens.Any.  Builds a parens.Symbol if nil.
+func WithSymbolFactory(f func(string) (parens.Any, error)) Option {
```

Comment:
why not use the same pattern as `WithNumReader` ? i.e., Add a `WithSymbolReader` that adds a custom reader macro and use this to override the default `readSymbol`?

Reply:
👍 Good call.

Reply:
@spy16  On second thought, I'm not too sure how to handle the [predefined symbol lookup](https://github.com/spy16/parens/blob/master/reader/reader.go#L354-L363).  I see two choices:

1. Export `reader.predef` so that custom `readSymbol` macros can access it (e.g. via a `Resolve` method).
2. Use a `func(string) Any` instead of a macro, so that we can build a symbol from a raw `string` token if it doesn't match any entries in `predef` (this is essentially what I have right now).

Thoughts?

Reply:
I've gone ahead and implemented option 1.  Note that this also adds a `Reader.Resolve()` method.  If you end up preferring option 2, it's just a simple matter of `git revert`.  🙂

Reply:
I see a 3rd option actually. 🤔 We remove the pre-defined symbols table from reader altogether and let the custom `symReader` handle it. what do you think? (predef existed as a way to customise some aspect of the symbol reading. but now that we allow customising the symbol reader itself, we can get rid of it i think. Just one less responsibility and smaller API surface for reader)

Reply:
Added a slightly different approach (Just to avoid adding a public Resolve() method to Reader and reduce reader responsibilities.).. Let me know if you see any issues.

Reply:
Yep, agree this is better.  Merging.